diagnostics/collector.go

```diff
@@ -24,29 +43,24 @@ const (
 
 var errNoColumnsFound = errors.New("no columns found")
 
-// NewDBCollector accepts database to work with during collect
-func NewDBCollector(adb db.Database) *DBCollector {
-	return &DBCollector{
-		db: adb,
-	}
-}
-
-func (c *DBCollector) Collect(ctx context.Context, w io.Writer) (err error) {
+// Collect query information from the database into csv files and write everything to io writer
+func (c *Collector) Collect(ctx context.Context, w io.Writer) (err error) {
 	writer := zip.NewWriter(w)
 	defer func() { err = writer.Close() }()
 
-	errs := c.collectWithZip(ctx, writer)
+	errs := c.collectDB(ctx, writer)
 
-	// write all errors into log
-	if err := writeErrors(writer, errs); err != nil {
-		return fmt.Errorf("write errors: %w", err)
+	rangeErrs := c.rangesStats(ctx, writer)
+	errs = append(errs, rangeErrs...)
+	if err = writeErrors(writer, errs); err != nil {
+		err = fmt.Errorf("write errors: %w", err)
 	}
 
 	return nil
 }
 
-// Collect query information from the database into csv files and write everything to io writer
-func (c *DBCollector) collectWithZip(ctx context.Context, writer *zip.Writer) []error {
+// collectDB query information from the database into csv files and write everything to io writer
+func (c *Collector) collectDB(ctx context.Context, writer *zip.Writer) []error {
```

Comment:
We can stop collecting information from `catalog_*` tables

Reply:
Done.